dbms/src/Interpreters/Join.cpp

```diff
@@ -1575,6 +1605,9 @@ void Join::joinBlockImpl(Block & block, const Maps & maps) const
 
     size_t rows = block.rows();
 
+    // If the probe block size is greater than max_block_size, we will set max_block_size to the probe block size to avoid some unnecessary split.
+    probe_process_info.max_block_size = std::max(probe_process_info.max_block_size, rows);
```

Comment:
hou about moving to `ProbeProcessInfo::setAndInit`.

Reply:
> hou about moving to `ProbeProcessInfo::setAndInit`.

done.